data/crac/crac-impl/src/test/java/com/farao_community/farao/data/crac_impl/HvdcRangeActionImplTest.java

```diff
@@ -0,0 +1,162 @@
+/*
+ * Copyright (c) 2019, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ *  License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+package com.farao_community.farao.data.crac_impl;
+
+import com.farao_community.farao.commons.FaraoException;
+import com.farao_community.farao.data.crac_api.Crac;
+import com.farao_community.farao.data.crac_api.Instant;
+import com.farao_community.farao.data.crac_api.range_action.*;
+import com.farao_community.farao.data.crac_api.usage_rule.UsageMethod;
+import com.farao_community.farao.data.crac_impl.utils.NetworkImportsUtil;
+import com.powsybl.iidm.network.Country;
+import com.powsybl.iidm.network.Network;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.util.Optional;
+import java.util.Set;
+
+import static org.junit.Assert.*;
+
+/**
+ * @author Godelaine de Montmorillon {@literal <godelaine.demontmorillon at rte-france.com>}
+ */
+public class HvdcRangeActionImplTest {
+    private Crac crac;
+    private HvdcRangeActionAdder hvdcRangeActionAdder;
+    private String networkElementId;
+    private Network network;
+
+    @Before
+    public void setUp() {
+        crac = new CracImplFactory().create("cracId");
+        network = NetworkImportsUtil.import16NodesNetworkWithHvdc();
+        networkElementId = "BBE2AA11 FFR3AA11 1";
+
+        hvdcRangeActionAdder = crac.newHvdcRangeAction()
+            .withId("hvdc-range-action-id")
+            .withName("hvdc-range-action-name")
+            .withNetworkElement("BBE2AA11 FFR3AA11 1")
+            .withOperator("operator")
+            .newFreeToUseUsageRule().withInstant(Instant.PREVENTIVE).withUsageMethod(UsageMethod.AVAILABLE).add();
+    }
+
+    @Test
+    public void apply() {
+        HvdcRangeAction hvdcRa = hvdcRangeActionAdder.newHvdcRange()
+                .withMin(-5)
+                .withMax(10)
```

Comment:
For this test case and the next, i find quite strange to define a Range [-5, 10], and to apply setpoint outside of this Range. Maybe we should throw an exception in that case (but I'm not convinced :p)